source/common/buffer/buffer_impl.cc

```diff
@@ -41,6 +41,19 @@ void OwnedImpl::commit(RawSlice* iovecs, uint64_t num_iovecs) {
   UNREFERENCED_PARAMETER(rc);
 }
 
+void OwnedImpl::copyOut(void* data, uint64_t size, size_t start) const {
+  ASSERT(start + size <= length());
+
+  evbuffer_ptr start_ptr;
+  int rc = evbuffer_ptr_set(buffer_.get(), &start_ptr, start, EVBUFFER_PTR_SET);
+  ASSERT(rc != -1);
```

Comment:
I guess I don't understand the failure mode for this function, but seem OK?

Reply:
Can't fail with proper input params which we verify.